torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -660,7 +668,7 @@ void Reducer::initialize_buckets(
           TORCH_CHECK(
               variable_index < replicas_[replica_index].size(),
               "Out of range variable index specified.");
-          const auto& variable = replicas_[replica_index][variable_index];
+          auto& variable = replicas_[replica_index][variable_index];
```

Comment:
forgot to revert it back, I was trying to make its grad point to an empty tensor, so that we can free the old bucket before allocating new bucket.  This does not work if users did not zero out grad or set grad to be none in their user codes, I'm still working on this part locally